pennylane/wires.py

```diff
@@ -0,0 +1,369 @@
+
+        Returns:
+            List: index list
+        """
+
+        return [self.index(w) for w in wires]
+
+    def subset(self, indices, periodic_boundary=False):
+        """
+        Returns a new Wires object which is a subset of this Wires object. The wires of the new
+        object are the wires at positions specified by 'indices'. Also accepts a single index as input.
+
+        For example:
+
+        >>> wires = Wires([4, 0, 1, 5, 6])
+        >>> wires.subset([2, 3, 0])
+        <Wires = [1, 5, 4]>
+
+        >>> wires.subset(1)
+        <Wires = [0]>
+
+        Args:
+            indices (List[int] or int): indices or index of the wires we want to select
+            periodic_boundary (bool): Whether the modulo of the number of wires of an index is used instead of an index.
+                Implements periodic boundary conditions in the indexing,
+                so that for example ``wires.subset(len(wires)) == wires.subset(0)``.
+
+        Returns:
+            Wires: subset of wires
+        """
+
+        if isinstance(indices, int):
+            indices = [indices]
+
+        if periodic_boundary:
+            # replace indices by their modulo
+            indices = [i % len(self.wire_tuple) for i in indices]
+
+        for i in indices:
+            if i > len(self.wire_tuple):
+                raise WireError(
+                    "Cannot subset wire at index {} from {} wires.".format(i, len(self.wire_tuple))
+                )
+
+        subset = [self.wire_tuple[i] for i in indices]
+        return Wires(subset)
+
+    def select_random(self, n_samples, seed=None):
+        """
+        Returns a randomly sampled subset of Wires of length 'n_samples'.
+
+        Args:
+            n_samples (int): number of subsampled wires
+            seed (int): optional random seed used for selecting the wires
+
+        Returns:
+            Wires: random subset of wires
+        """
+
+        if n_samples > len(self.wire_tuple):
+            raise WireError(
+                "Cannot sample {} wires from {} wires.".format(n_samples, len(self.wire_tuple))
+            )
+
+        if seed is not None:
+            np.random.seed(seed)
+
+        indices = np.random.choice(len(self.wire_tuple), size=n_samples, replace=False)
+        subset = [self.wire_tuple[i] for i in indices]
+        return Wires(subset)
+
+    @staticmethod
+    def shared_wires(list_of_wires):
+        """Return only the wires that appear in each Wires object in the list.
+
+        This is similar to a set intersection method, but keeps the order of wires as they appear in the list.
+
+        For example:
+
+        >>> wires1 =  Wires([4, 0, 1])
+        >>> wires2 = Wires([3, 0, 4])
+        >>> wires3 = Wires([4, 0])
+        >>> Wires.shared_wires([wires1, wires2, wires3])
+        <Wires = [4, 0]>
+
+        >>> Wires.shared_wires([wires2, wires1, wires3])
+        <Wires = [0, 4]>
```

Comment:
Perfect example, answers the question I was just about to ask!
```suggestion
        >>> wires1 =  Wires([4, 0, 1])
        >>> wires2 = Wires([3, 0, 4])
        >>> wires3 = Wires([4, 0])
        >>> Wires.shared_wires([wires1, wires2, wires3])
        <Wires = [4, 0]>
        >>> Wires.shared_wires([wires2, wires1, wires3])
        <Wires = [0, 4]>
```